corehq/apps/reports/filters/api.py

```diff
@@ -197,3 +223,187 @@ def get(self, request, domain):
 
 class DeviceLogIds(DeviceLogFilter):
     field = 'device_id'
+
+
+@require_POST
+@toggles.COPY_CASES.required_decorator()
+@require_permission(HqPermissions.edit_data)
+@requires_privilege(privileges.CASE_COPY)
+@location_safe
+def copy_cases(request, domain, *args, **kwargs):
```

Comment:
nit: this file is `reports/filters/api.py`, which is used for providing lists of options to dropdown filters.  This functionality probably belongs elsewhere.